projects/RabbitMQ.Client/client/impl/RabbitMQActivitySource.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using RabbitMQ.Client.Events;
+using RabbitMQ.Client.Framing.Impl;
+
+namespace RabbitMQ.Client
+{
+    internal static class RabbitMQActivitySource
+    {
+        internal const string MessageId = "messaging.message.id";
+        internal const string MessageConversationId = "messaging.message.conversation_id";
+        internal const string MessagingOperation = "messaging.operation";
+        internal const string MessagingSystem = "messaging.system";
+        internal const string MessagingDestination = "messaging.destination.name";
+        internal const string MessagingDestinationRoutingKey = "messaging.rabbitmq.destination.routing_key";
+        internal const string MessagingPayloadSize = "messaging.message.payload_size_bytes";
+        internal const string ProtocolName = "net.protocol.name";
+        internal const string ProtocolVersion = "net.protocol.version";
+
+
+        internal static readonly ActivitySource s_source = new ActivitySource("RabbitMQ.Client", typeof(RabbitMQActivitySource).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion);
+
+        internal static Activity Send(string routingKey, string exchange, int bodySize)
+        {
+            if (s_source.HasListeners())
+            {
+                Activity activity = StartRabbitMQActivity($"{routingKey} publish", ActivityKind.Producer, Activity.Current?.Context ?? default);
+                if (activity?.IsAllDataRequested == true)
+                {
+                    PopulateMessagingTags("publish", routingKey, exchange, 0, null, bodySize, activity);
+                }
+
+                return activity;
+            }
+
+            return null;
+        }
+
+        internal static Activity Receive(string routingKey, string exchange, ulong deliveryTag, IReadOnlyBasicProperties readOnlyBasicProperties, int bodySize)
+        {
+            if (s_source.HasListeners())
+            {
+                // Extract the PropagationContext of the upstream parent from the message headers.
+                DistributedContextPropagator.Current.ExtractTraceIdAndState(readOnlyBasicProperties, ExtractTraceIdAndState, out string traceId, out string traceState);
```

Comment:
nit: `DistributedContextPropagator` unfortunately uses a wrong term - it should be `traceparent` (i.e. 00-traceId-spanId-01), not `traceId`  
```suggestion
                DistributedContextPropagator.Current.ExtractTraceIdAndState(readOnlyBasicProperties, ExtractTraceIdAndState, out string traceparent, out string traceState);
```